src/updater/updater.cpp

```diff
@@ -120,10 +120,16 @@ auto mmapFile(const std::string &filename, boost::interprocess::mode_t mode)
     using boost::interprocess::file_mapping;
     using boost::interprocess::mapped_region;
 
+    {
+        storage::io::FileReader file(filename, storage::io::FileReader::VerifyFingerprint);
```

Comment:
> or do an implicit fclose on the destructor

Wait we don't close file handles? This is definitely a bug that should be fixed.

/cc @danpat I think these might be the reason we saw the file handle exhaustion.